### PR TITLE
Some slapcrafting stuff

### PR DESCRIPTION
**Pull request by Pebblesthe2nd**
## About The Pull Request

Made a couple of slapcrafting things. so stuff can be done directly with in game items without menus
- Cleaning & Heating cosmic knife (welder only for now)
- Cutting clothing into cloth rags
- Recycling Combat armor with a saw
- Recycling Power armor with a blowtorch

Some minor fixes for smithing, sappers get a modest stamina damage for now to make them a tiny bit less useless since they are cool.

## Pre-Merge Checklist

- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.

## Changelog
:cl:
code: added slapcrafting for cosmic knife, recycling PA and CA, cutting clothing.
/:cl:

### CODE REVIEW

File: code/game/objects/items/melee/f13onehanded.dm [hunk 1]
```diff
@@ -274,24 +274,52 @@ obj/item/melee/onehanded/knife/switchblade
...
+		user.visible_message("[user] begins cleaning the blade of the [src].", \
+				"<span class='notice'>You begin scrubbing the [src] with Abraxo. Smells nice.</span>", \
+				"<span class='italics'>You hear faint sounds of scrubbing.</span>")
+		playsound(get_turf(src), 'sound/FermiChem/heatdam.ogg', 50, TRUE)
+		if(!do_after(user, 60, TRUE, src))
+			return
+		new /obj/item/melee/onehanded/knife/cosmic(drop_location())
+		qdel(src)
+		qdel(C)
+	return TRUE
```
Comment (EgorDinamit) [inline code/game/objects/items/melee/f13onehanded.dm:302]:
```suggestion
		qdel(C)
		return
	return ..()
```

Comment (EgorDinamit) [reply code/game/objects/items/melee/f13onehanded.dm:302]: Same for all procs below.
Might also want to make it automatically put the new item in hand.

File: code/modules/clothing/head/f13factionhead.dm [hunk 2]
```diff
@@ -707,6 +707,21 @@
...
+// recycling the above
+/obj/item/clothing/head/helmet/f13/ncr/rangercombat/attackby(obj/item/I, mob/user, params)
+	if(I.tool_behaviour == TOOL_SAW)
+		user.visible_message("[user] begins recycling the [src].", \
+				"<span class='notice'>You begin recycling the [src].</span>", \
+				"<span class='italics'>You hear the noise of a saw cutting through metal and ceramic.</span>")
+		playsound(get_turf(src), 'sound/weapons/circsawhit.ogg', 50, TRUE)
+		if(!do_after(user, 50, TRUE, src))
+			return
+		new /obj/item/stack/crafting/armor_plate/ (drop_location(), 2)
```
Comment (EgorDinamit) [inline code/modules/clothing/head/f13factionhead.dm:719]: Just two..?

**Comment by EgorDinamit** [comment]
A good idea, but it needs some work, you also didn't remove existing crafting recipes yet.

**Comment by Pebblesthe2nd** [comment]
> 
The return was a good catch, two plates might be too stingy, maybe not, bumping it to three wont hurt, then I wash my hands of it. Included this in a update.
Placing the stuff inhand is something I only like for some specific things, not for this, not gonna add it.
As for salvaging, I assume you refer to the mechanics you just added? Tested it on a T-45D just now, recycling it mid salvage process worked just fine.

I'm aware the terrible old system remains, it will not be so for too long. For now, this is good to go.

**Pull request merged**